### PR TITLE
Infix to Prefix Converter in JavaScript

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#5083 )

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Sir please review it @ankitaggarwal23 @gat786 @siddharth25pandey @atarax665

File: JavaScript/README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@
 - [Recursive preorder traversal in binary search tree](./ds/Recursive_preorder.js)
 - [Recursive inorder traversal in binary search tree](./ds/Recursive_inorder.js)
 - [Recursive postorder traversal of binary search tree](./ds/Recursive_postorder.js)
+- [Infix to Prefix Conversion in JavaScript](./ds/infixtoprefix.js)
```
Comment (ankitaggarwal23) [inline JavaScript/README.md]: Rewrite it as  `[Infix to Prefix Conversion]`

File: JavaScript/ds/infixtoprefix.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+/*
+ This project converts infix expression to prefix expression in JavaScript.
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js]: This is not a project rewrite to code

File: JavaScript/ds/infixtoprefix.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            if (el =='(') {
+                while (stackarr[topp] != ")") {
+                  prefix[temp++] = pop();
+                }
+          pop();
+            }
+            else if(el==')')
+            {
+                push(el);
+            }
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js:89]: Align your code with proper format and remove redundant comments

File: JavaScript/ds/infixtoprefix.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    }    
+    console.log("Prefix Expression:- "+st);
+ }
+ InfixtoPrefix("((a+b)*c)");
+ 
+
+ /*
+OutPut:
+*+abc
+
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js]: Remove redundant comments and line

File: JavaScript/ds/infixtoprefix.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+Output:
++A*B/C-EF
+
+
+Input:
+a+b*(c^d-e)^(f+g*h)-i
+
+Output:
++a-*b^-^cde+f*ghi
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js]: - Write outputs with proper display message
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Add your code to repl.it and share the link as well.

**Comment by shivammahajancse** [comment]
Repl link: https://replit.com/join/ijzcusnn-shivammahajan4
All requested changes done @ankitaggarwal23 sir

File: JavaScript/ds/infixtoprefix.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    for(var i=infixval.length-1;i>=0;i--)
+    {
+        var el=infixval[i];
+        if(operator(el))
+        {
+            if (el =='(') {
+                while (stackarr[topp] != ")") {
+                  prefix[temp++] = pop();
+                }
+          pop();
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js]: Add a tab space before `pop`

File: JavaScript/ds/infixtoprefix.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        st+=prefix[j];
+    }    
+    console.log("Prefix Expression:- "+st);
+ }
+ InfixtoPrefix("((a+b)*c)");
+
+ /*
+OutPut:
+Prefix Expression: *+abc
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js]: Remove these lines

File: JavaScript/ds/infixtoprefix.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+A+B*C/(E-F)
+
+Output:
+Prefix Expression: +A*B/C-EF
+
+Input:
+a+b*(c^d-e)^(f+g*h)-i
+
+Output:
+Prefix Expression: +a-*b^-^cde+f*ghi
```
Comment (ankitaggarwal23) [inline JavaScript/ds/infixtoprefix.js]: Write proper display message while reading and printing the outputs

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
Please review @ankitaggarwal23 sir.

**Review by atarax665 — APPROVED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**